src/Microsoft.OData.Core/Uri/SelectExpandClauseToStringBuilder.cs

```diff
@@ -33,43 +28,51 @@ public string TranslateSelectExpandClause(SelectExpandClause selectExpandClause,
 
             List<string> selectList = selectExpandClause.GetCurrentLevelSelectList();
             string selectClause = null;
+            string expandClause = null;
+
             if (selectList.Any())
             {
                 selectClause = String.Join(ODataConstants.ContextUriProjectionPropertySeparator, selectList.ToArray());
             }
 
-            selectClause = string.IsNullOrEmpty(selectClause) ? null : string.Concat("$select", ExpressionConstants.SymbolEqual, isFirstSelectItem ? Uri.EscapeDataString(selectClause) : selectClause);
-            isFirstSelectItem = false;
-
-            string expandClause = null;
-            foreach (ExpandedNavigationSelectItem expandSelectItem in selectExpandClause.SelectedItems.Where(I => I.GetType() == typeof(ExpandedNavigationSelectItem)))
+            foreach (SelectItem selectItem in selectExpandClause.SelectedItems)
             {
-                if (string.IsNullOrEmpty(expandClause))
-                {
-                    expandClause = firstFlag ? expandClause : string.Concat("$expand", ExpressionConstants.SymbolEqual);
-                }
-                else
+                if(selectItem.GetType() == typeof(PathSelectItem))
                 {
-                    expandClause += ExpressionConstants.SymbolComma;
+                    selectClause += this.Translate((PathSelectItem)selectItem);
                 }
 
-                expandClause += this.Translate(expandSelectItem);
-            }
-
-            foreach (ExpandedReferenceSelectItem expandSelectItem in selectExpandClause.SelectedItems.Where(I => I.GetType() == typeof(ExpandedReferenceSelectItem)))
-            {
-                if (string.IsNullOrEmpty(expandClause))
+                if (selectItem.GetType() == typeof(ExpandedNavigationSelectItem))

```

Comment:
```suggestion
                else if (selectItem.GetType() == typeof(ExpandedNavigationSelectItem))
```
So this check is not done if first `if` block was executed...